src/runtime/flow_node_handler/boundary_event_handler/error_boundary_event_handler.ts

```diff
@@ -37,6 +38,10 @@ export class ErrorBoundaryEventHandler extends BoundaryEventHandler {
       (!errorDefinition.code || errorDefinition.code === '') ||
       errorDefinition.code === (error as BpmnError).code;
 
+    if (errorNamesMatch && errorCodesMatch) {
```

Comment:
Since this expression is now used twice, I think it makes sense to move it into a separate business rule.

```suggestion
    const boundaryEventCanHandleError = errorNamesMatch && errorCodesMatch;
    if (boundaryEventCanHandleError) {
```